src/main/java/net/dries007/tfc/common/blocks/crop/CropBlock.java

```diff
@@ -41,32 +40,39 @@
 import net.dries007.tfc.util.Helpers;
 import net.dries007.tfc.util.calendar.Calendars;
 import net.dries007.tfc.util.climate.ClimateRange;
+import net.dries007.tfc.util.registry.RegistryCrop;
 
 public abstract class CropBlock extends net.minecraft.world.level.block.CropBlock implements HoeOverlayBlock, ICropBlock, IForgeBlockExtension, EntityBlockExtension
 {
     public static final VoxelShape QUARTER_SHAPE = box(2, 0, 2, 14, 4, 14);
     public static final VoxelShape HALF_SHAPE = box(2, 0, 2, 14, 8, 14);
     public static final VoxelShape FULL_SHAPE = box(2, 0, 2, 14, 16, 14);
 
-    protected final FarmlandBlockEntity.NutrientType primaryNutrient;
+    protected final Supplier<FarmlandBlockEntity.NutrientType> primaryNutrient;
     protected final Supplier<? extends Block> dead;
     protected final Supplier<? extends Item> seeds;
     protected final Supplier<ClimateRange> climateRange;
+    protected final Supplier<Integer> baseGrowthTime;
+    protected final Supplier<Integer> nutrientConsumptionTime;
+    protected final Supplier<Double> nutrientResupplyFactor;
 
     private final ExtendedProperties extendedProperties;
     private final int maxAge;
 
-    protected CropBlock(ExtendedProperties properties, int maxAge, Supplier<? extends Block> dead, Supplier<? extends Item> seeds, FarmlandBlockEntity.NutrientType primaryNutrient, Supplier<ClimateRange> climateRange)
+    protected CropBlock(ExtendedProperties properties, int maxAge, RegistryCrop crop)
```

Comment:
More unnecessary breaking changes. Again, what is the motivation for this? Is there a reason? Because I don't see one.